client/app/scripts/charts/nodes-chart-elements.js

```diff
@@ -1,30 +1,236 @@
+    const targetNoMatches = searchNodeMatches.get(edge.get('target'), makeMap()).isEmpty();
+    const notMatched = this.props.searchQuery && (sourceNoMatches || targetNoMatches);
+    const sourceInNetwork = selectedNetworkNodesIds.contains(edge.get('source'));
+    const targetInNetwork = selectedNetworkNodesIds.contains(edge.get('target'));
+    const notInNetwork = this.props.selectedNetwork && (!sourceInNetwork || !targetInNetwork);
+    return edge.set('blurred', !edge.get('highlighted') && !edge.get('focused') &&
+      (otherNodesSelected || notMatched || notInNetwork));
+  }
+
+  edgeScaleDecorator(edge) {
+    return edge.set('scale', edge.get('focused') ? this.props.selectedScale : 1);
+  }
+
+  renderNode(node) {
+    const { isAnimated, contrastMode } = this.props;
+    return (
+      <NodeContainer
+        matches={node.get('matches')}
+        networks={node.get('networks')}
+        metric={node.get('metric')}
+        blurred={node.get('blurred')}
+        focused={node.get('focused')}
+        highlighted={node.get('highlighted')}
+        shape={node.get('shape')}
+        stack={node.get('stack')}
+        key={node.get('id')}
+        id={node.get('id')}
+        label={node.get('label')}
+        labelMinor={node.get('labelMinor')}
+        pseudo={node.get('pseudo')}
+        rank={node.get('rank')}
+        dx={node.get('x')}
+        dy={node.get('y')}
+        scale={node.get('scale')}
+        isAnimated={isAnimated}
+        contrastMode={contrastMode}
+      />
+    );
+  }
+
+  renderEdge(edge) {
+    const { isAnimated } = this.props;
+    return (
+      <EdgeContainer
+        key={edge.get('id')}
+        id={edge.get('id')}
+        source={edge.get('source')}
+        target={edge.get('target')}
+        waypoints={edge.get('points')}
+        highlighted={edge.get('highlighted')}
+        focused={edge.get('focused')}
+        blurred={edge.get('blurred')}
+        scale={edge.get('scale')}
+        isAnimated={isAnimated}
+      />
+    );
+  }
+
+  renderElement(element) {
+    // This heuristics is not ideal but it works.
+    return element.get('points') ? this.renderEdge(element) : this.renderNode(element);
+  }
+
   render() {
-    const { layoutNodes, layoutEdges, selectedScale, isAnimated } = this.props;
+    const nodes = this.props.layoutNodes.toIndexedSeq()
+      .map(this.nodeHighlightedDecorator)
+      .map(this.nodeFocusedDecorator)
+      .map(this.nodeBlurredDecorator)
+      .map(this.nodeMatchesDecorator)
+      .map(this.nodeNetworksDecorator)
+      .map(this.nodeMetricDecorator)
+      .map(this.nodeScaleDecorator)
+      .groupBy(this.nodeDisplayLayer);
+
+    const edges = this.props.layoutEdges.toIndexedSeq()
+      .map(this.edgeHighlightedDecorator)
+      .map(this.edgeFocusedDecorator)
+      .map(this.edgeBlurredDecorator)
+      .map(this.edgeScaleDecorator)
+      .groupBy(this.edgeDisplayLayer);
+
+    // NOTE: The elements need to be arranged into a single array outside
```

Comment:
I was trying to find some reference to it but couldn't :/

I figured it was true by first having the lists in the DOM and noticing that the `render` method was being invoked for too many nodes. Then I just tried making one flat list outside and it worked..